include/motion_control/instance_list_node.h

```diff
@@ -0,0 +1,65 @@
+#ifndef MOTION_CONTROL_INSTANCE_LIST_NODE_H
+#define MOTION_CONTROL_INSTANCE_LIST_NODE_H
+
+namespace MotionControl
+{
+
+template <class T> class InstanceLinkedList;
+
+template <class T>
+class InstanceListNode
+{
+    friend class InstanceLinkedList<T>;
+public:
+    InstanceListNode(T element);
+    ~InstanceListNode();
+
+    InstanceListNode<T>* get_next();
+    InstanceListNode<T>* get_prev();
+    bool in_list();
```

Comment:
We had a discussion earlier about making some sort of iterator for this. Not necessary to change it now, but let's put it on the backburner for possible future development.